File: 01-ARRAYS/remove-duplicates-from-sorted-array.py
from typing import *


# https://leetcode.com/problems/remove-duplicates-from-sorted-array/
def removeDuplicates(nums: List[int]) -> int:
    uniquesLength = 1
    for i in range(1, len(nums)):
        if nums[i - 1] != nums[i]:
            nums[uniquesLength] = nums[i]
            uniquesLength += 1

    return uniquesLength

File: 01-ARRAYS/Go/remove-duplicates-from-sorted-array.go
package main

// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
func removeDuplicates(nums []int) int {
	k := 1
	for i := 1; i < len(nums); i++ {
		if nums[i-1] != nums[i] {
			nums[k] = nums[i]
			k++
		}
	}

	return k
}
